proposals/utils/proposal_utils.py

```diff
@@ -373,3 +377,60 @@ def notify_local_staff(proposal):
 
     # Reset the current language
     activate(current_language)
+
+
+@deconstructible
+class FilenameFactory:
+    '''A callable class which can be passed to upload_to() in FileFields
+    and can be deconstructed for migrations'''
+    
+    def __init__(self, document_type):
+        self.document_type = document_type
+    
+    def __call__(self, instance, original_fn):
+        '''Returns a custom filename preserving the original extension,
+        something like "FETC-2020-002-01-Villeneuve-T2-Informed-Consent.pdf"'''
+        
+        # Importing here to prevent circular import
+        from proposals.models import Proposal
+        
+        if isinstance(instance, Proposal):
+            proposal = instance
+            trajectory = None
+        else:
+            # In case of Documents or Study objects
+            proposal = instance.proposal
+            try:
+                trajectory = 'T' + str(instance.study.id)
```

Comment:
This should be ``instance.study.order``

Reply:
(ID results in large numbers that don't make sense for the reviewer, as it's the PK. ``order`` is the trajectory number specific for this proposal)

Reply:
Ah yes, thank you for that. Locally I only had at most a couple proposals going and therefore didn't notice